File: 26. Remove Duplicates from Sorted Array.py
# Will fail if first element is X. Simpler solution: return sorted(list(set(nums)))

class Solution:
    def removeDuplicates(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        h = 'X'
        p = 0

        for i in range(0, len(nums)):
            if h == nums[p]:
                nums.pop(p)
            else:
                h = nums[p]
                p += 1

        return len(nums)
            
print(Solution().removeDuplicates([0,0,1,1,10,0,0,0,0]))
        

File: 26. Remove Duplicates from Sorted Array.cpp
// More efficient than using pop / erase to remove the item physically

class Solution {
public:
    int removeDuplicates(vector<int>& nums) {
        
        int j = 0;
        
        if (nums.size() == 0) {
            return 0;
        }
        
        for (int i = 0; i < nums.size(); i++) {
                        
            if (nums[i] != nums[j]) {
                j++;
            	nums[j] = nums[i];
            }
        }
        
        return j+1;
    }
};